shell/platform/windows/game_pad_winuwp.cc

```diff
@@ -0,0 +1,228 @@
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/game_pad_winuwp.h"
+
+namespace flutter {
+
+GamepadWinUWP::GamepadWinUWP(GamepadDualAxisCallback leftstick,
+                             GamepadDualAxisCallback rightstick,
+                             GamepadSingleAxisCallback lefttrigger,
+                             GamepadSingleAxisCallback righttrigger,
+                             GamepadButtonCallback pressedcb,
+                             GamepadButtonCallback releasedcb,
+                             GamepadAddedRemovedCallback changedcb)
+    : pressed_buttons_(winrt::Windows::Gaming::Input::GamepadButtons::None),
+      left_stick_x_value_(0),
+      left_stick_y_value_(0),
+      right_stick_x_value_(0),
+      right_stick_y_value_(0),
+      left_trigger_value_(0),
+      right_trigger_value_(0) {
+  left_stick_callback_ = leftstick;
+  right_stick_callback_ = rightstick;
+
+  left_trigger_callback_ = lefttrigger;
+  right_trigger_callback_ = righttrigger;
+
+  button_pressed_callback_ = pressedcb;
+  button_released_callback_ = releasedcb;
+
+  arrival_departure_callback_ = changedcb;
+}
+
+void GamepadWinUWP::Initialize() {
+  RefreshCachedGamepads();
+
+  winrt::Windows::Gaming::Input::Gamepad::GamepadAdded(
+      {this, &GamepadWinUWP::OnGamepadAdded});
+  winrt::Windows::Gaming::Input::Gamepad::GamepadRemoved(
+      {this, &GamepadWinUWP::OnGamepadRemoved});
+
+  const std::lock_guard<std::mutex> lock(gamepad_mutex_);
+  current_game_pad_ = GetLastGamepad();
+}
+
+bool GamepadWinUWP::HasController() {
+  const std::lock_guard<std::mutex> lock(gamepad_mutex_);
+  return current_game_pad_ != nullptr;
+}
+
+void GamepadWinUWP::OnGamepadAdded(
+    winrt::Windows::Foundation::IInspectable const& th,
+    winrt::Windows::Gaming::Input::Gamepad const& args) {
+  enumerated_game_pads_.push_back(args);
+
+  {
+    const std::lock_guard<std::mutex> lock(gamepad_mutex_);
+    auto most_recent = GetLastGamepad();
+    if (current_game_pad_ != most_recent) {
+      current_game_pad_ = most_recent;
+    }
+  }
+
+  if (this->arrival_departure_callback_ != nullptr) {
+    arrival_departure_callback_();
+  }
+}
+
+void GamepadWinUWP::OnGamepadRemoved(
+    winrt::Windows::Foundation::IInspectable const&,
+    winrt::Windows::Gaming::Input::Gamepad const& /*args*/) {
+  RefreshCachedGamepads();
+}
+
+void GamepadWinUWP::RefreshCachedGamepads() {
+  enumerated_game_pads_.clear();
+  auto gamepads = winrt::Windows::Gaming::Input::Gamepad::Gamepads();
+  for (auto gamepad : gamepads) {
+    enumerated_game_pads_.push_back(gamepad);
+  }
+}
+
+const winrt::Windows::Gaming::Input::Gamepad* GamepadWinUWP::GetLastGamepad() {
+  winrt::Windows::Gaming::Input::Gamepad* gamepad = nullptr;
+
+  if (!enumerated_game_pads_.empty()) {
+    gamepad = &enumerated_game_pads_.back();
+  }
+
+  return gamepad;
+}
+
+bool isValid(double value) {
+  if (value > 0.1 || value < -0.1) {
+    return true;
+  }
+  return false;
```

Comment:
nit: just `return value > 0.1 || value < -0.1;`

Reply:
Fixed in next commit